### PR TITLE
application going to welcome when disconnect wallet, removed holesky from claim, fixed bug when change metamask account

### PR DESCRIPTION
**Pull request by axelrod-blox**

### CODE REVIEW

**Review by meshin-dev — APPROVED** [review]

**Review by alexstolr — APPROVED** [review]

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
             \"explorerUrl\": \"https://explorer.stage.ssv.network\",
             \"googleTagSecret\": \"${{ secrets.STAGE_GOOGLE_TAG_SECRET }}\",
             \"tokenAddress\": \"0x6471F70b932390f527c6403773D082A0Db8e8A9F\",
+            \"distributionContract\": \""${{ secrets.STAGE_DISTRIBUTION_CONTRACT_ADDRESS }}\",
```
Comment (meshin-dev) [inline .github/workflows/build_deploy.yml:28]: Remove one `"`

File: .github/workflows/build_deploy.yml [hunk 2]
```diff
@@ -49,6 +50,7 @@ jobs:
             \"api\": \"https://api.ssv.network/api\",
             \"explorerUrl\": \"https://explorer.ssv.network\",
             \"googleTagSecret\": \"${{ secrets.PROD_GOOGLE_TAG_SECRET }}\",
+            \"distributionContract\": \""${{ secrets.PROD_DISTRIBUTION_CONTRACT_ADDRESS_MAINNET }}\",
```
Comment (meshin-dev) [inline .github/workflows/build_deploy.yml:53]: Here too

File: .github/workflows/build_deploy.yml [hunk 3]
```diff
@@ -60,6 +62,7 @@ jobs:
             \"api\": \"https://api.ssv.network/api\",
             \"explorerUrl\": \"https://goerli.explorer.ssv.network\",
             \"googleTagSecret\": \"${{ secrets.PROD_GOOGLE_TAG_SECRET }}\",
+            \"distributionContract\": \""${{ secrets.PROD_DISTRIBUTION_CONTRACT_ADDRESS_TESTNET }}\",
```
Comment (meshin-dev) [inline .github/workflows/build_deploy.yml:65]: And here

**Review by meshin-dev — CHANGES_REQUESTED** [review]

File: .github/workflows/build_deploy.yml (hunk 1 above)
Comment (axelrod-blox) [reply .github/workflows/build_deploy.yml:28]: done

File: src/app/hooks/useDistributionStore.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  const stores = useStores();
+  const mainnetStore: DistributionStore = stores.Distribution;
+  const testnetStore: DistributionTestnetStore = stores.DistributionTestnet;
+  const enableTestingFlow = getLocalStorageFlagValue(DEVELOPER_FLAGS.ENABLE_TEST_DISTRIBUTION_FLOW);
+  const currentFlow = enableTestingFlow ? enableTestingFlow : networkId;
+
+  const distributionStores = {
+    [NETWORKS.GOERLI]: testnetStore,
+    [NETWORKS.MAINNET]: mainnetStore,
+  };
```
Comment (meshin-dev) [inline src/app/hooks/useDistributionStore.ts:17]: Well, i see mainnet and goerly (old ot new?)
We should have support for old goerli and new maknnet. 

But in addition to that, we should have ABI equal to mainnet to work with for new goerli distribution (for testing purposes). 

Please explain how to achieve this. 
In your task please

Comment (axelrod-blox) [reply src/app/hooks/useDistributionStore.ts:17]: we have additional distributionHelper, which return contract and ABI by network ID, but, we also have there some logic which check future flag for testing, and if he exist he return the abi which equal to mainnet and also contract addres which we will use to testing, contract address for testing also should be provide in local storage

**Review by meshin-dev — APPROVED** [review]

**Pull request merged**